### PR TITLE
Doc - 3.2.4 Mathematical + 2.4 Subjects

### PR DESCRIPTION
**Pull request by at-linhvo**
- [x] concat

- [x] toArray

- [x] reduce

- [x] Subjects

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -345,6 +345,154 @@ observable.filter { $0.hasPrefix("Number") } // 2
 2. Lọc nội dụng bắt đầu bằng chuỗi `Number`
 3. Subcrible một observable để có thể xử lý mỗi khi nội dung search bar thay đổi
 
+### 2.4. Subjects
+
+​	Trong Reactive Programming, Observable là nơi khởi nguồn, là gốc của  mọi stream. Nhưngcó những trường hợp, chúng ta không thể đơn thuần bọc một đối tượng bên trong một Observable, sau đó phát ra những tín hiệu.
```
Comment (at-thinhung) [inline README.md]: @at-linhvo 
Ghi ngắn gọn, dễ hiểu hơn tí

File: README.md [hunk 2]
```diff
@@ -1089,6 +1237,51 @@ Chạy code trên ta được điều cần chứng minh: :D
 
 #### 3.2.4. Mathematical
 
+- **concat**
```
Comment (at-thinhung) [inline README.md]: @at-linhvo 
Remove

File: README.md [hunk 3]
```diff
@@ -345,6 +345,154 @@ observable.filter { $0.hasPrefix("Number") } // 2
...
+
+​	Chẳng hạn khi sử dụng UIImagePickerController, ngoài việc quan tâm tới các hình ảnh mà người dùng chọn (stream), ứng dụng cần tương tác với chính UIImagePickerController để ẩn, hiển, … như vậy không thể bọc UIImagePickerController bên trong Observable. Khi đó, Subject sẽ đóng vai trò cầu nối, giúp chuyển đổi các tương tác của người dùng thành các Observable tương ứng.
+
+#### 2.4.1. PublishSubject
+
+​	PublishSubject là các phần tử có thể được phát ngay sau khi Subject được khởi tạo, bất chấp chưa có đối tượng nào subscribe tới nó (hot observable). Observer sẽ không nhận được các phần tử phát ra trước thời điểm subscribe.
+
+![PublishSubject-diagram](./resources/images/2.4/PublishSubject-diagram.png)
+
+```swift
```
Comment (at-suho) [inline README.md]: Please add comments

Comment (at-linhvo) [reply README.md]: Done 👍

**Pull request merged**